src/dtemplate.d

```diff
@@ -4105,6 +4109,44 @@ extern (C++) MATCH deduceType(RootObject o, Scope* sc, Type tparam, TemplatePara
                 result = match2;
                 return;
             }
+
+            /* Deduce common type
```

Comment:
Yes, basically we need to do it, but current existing common-type deduction code `typeMerge` does not have convenient interface to reuse it in here.